staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -615,61 +623,40 @@ func TestNodeChangesInExternalLoadBalancer(t *testing.T) {
 			},
 			worker:                3,
 			nodeListerErr:         nil,
-			expectedRetryServices: []*v1.Service{},
+			expectedRetryServices: sets.NewString(),
 		},
 		{
 			desc:                  "error occur during sync",
 			nodes:                 []*v1.Node{node1, node2, node3, node4},
 			expectedUpdateCalls:   []fakecloud.UpdateBalancerCall{},
 			worker:                3,
 			nodeListerErr:         fmt.Errorf("random error"),
-			expectedRetryServices: services,
+			expectedRetryServices: serviceNames,
 		},
 		{
 			desc:                  "error occur during sync with 1 workers",
 			nodes:                 []*v1.Node{node1, node2, node3, node4},
 			expectedUpdateCalls:   []fakecloud.UpdateBalancerCall{},
 			worker:                1,
 			nodeListerErr:         fmt.Errorf("random error"),
-			expectedRetryServices: services,
+			expectedRetryServices: serviceNames,
 		},
 	} {
 		t.Run(tc.desc, func(t *testing.T) {
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 			controller.nodeLister = newFakeNodeLister(tc.nodeListerErr, tc.nodes...)
-			servicesToRetry := controller.updateLoadBalancerHosts(ctx, services, tc.worker)
-			compareServiceList(t, tc.expectedRetryServices, servicesToRetry)
+			servicesToRetry := sets.NewString()
+			for key := range controller.updateLoadBalancerHosts(ctx, services, tc.worker) {
```

Comment:
Because `updateLoadBalancerHosts` returns a sets.String already, I think you can simplify this to just:
```
servicesToRetry := controller.updateLoadBalancerHosts(ctx, services, tc.worker)
assert.Truef(t, tc.expectedRetryServices.Equal(servicesToRetry), "Services to retry are not expected")
```

Reply:
You are right. Updated.